apps/myaccount/src/components/profile/profile.tsx

```diff
@@ -1249,8 +1304,11 @@ export const Profile: FunctionComponent<ProfileProps> = (props: ProfileProps): R
                             || schema.name === ProfileConstants?.SCIM2_SCHEMA_DICTIONARY.get("RESROUCE_TYPE")
                             || schema.name === ProfileConstants?.SCIM2_SCHEMA_DICTIONARY.get("EXTERNAL_ID")
                             || schema.name === ProfileConstants?.SCIM2_SCHEMA_DICTIONARY.get("META_DATA")
-                            || (!commonConfig.userProfilePage.showEmail &&
-                                schema.name === ProfileConstants?.SCIM2_SCHEMA_DICTIONARY.get("EMAILS"))
+                            || ((
+                                !commonConfig.userProfilePage.showEmail
+                                || usernameConfig?.enableValidator==="false"
+                            )
+                                && schema.name === ProfileConstants?.SCIM2_SCHEMA_DICTIONARY.get("EMAILS"))
```

Comment:
WDYT about moving this whole if check to a separate function.

Reply:
Since this is a set of if block conditions, I feel it is not necessary to create a separate function for this.